docs/experimental-code.md

```diff
@@ -31,39 +38,109 @@ const unstable_meta = {
 
 // An unstable component will retain its name, specifically for things like
 // the rules of hooks plugin which depend on the correct casing of the name
-function Pagination(props) {
+function StaticNotification(props) {
   // ...
 }
 
 // However, when we export the component we will export it with the `unstable_`
 // prefix. (Similar to React.unstable_Suspense, React.unstable_Profiler)
-export { default as unstable_Pagination } from './components/Pagination';
+export { default as unstable_StaticNotification } from './components/StaticNotification';
 ```
 
 For teams using these features, they will need to import the functionality by
 using the `unstable_` prefix. For example:
 
 ```jsx
-import { unstable_Pagination as Pagination } from 'carbon-components-react';
+import { unstable_StaticNotification as StaticNotification } from '@carbon/react';
 ```
 
-## Experimental status
+### Documenting components and exports prefixed with `unstable_`
+
+Within the experimental section of the `@carbon/react` storybook, there are
+individual entries for each unstable export. These contain stories and written
+documentation outlining the functionality of these components and exports.
+
+## Feature flags
+
+Some Carbon packages ship with available feature flags. These feature flags
+enable new behavior and styling, allowing you to opt-in to new (and sometimes
+breaking) changes while using the current major version. A feature flag may be
+configured in javascript, sass, or both.
+
+All the currently available feature flags, as well as how to enable them, is
+documented in the
+[`@carbon/react` storybook](https://react.carbondesignsystem.com/?path=/docs/experimental-feature-flags--overview).
+
+### Feature flags are opt-in
+
+Feature flags are typically marked `false` or "off" by default to ensure
+backwards compatibility. These flags can be turned on individually within a
+consuming project to allow opt-in of the features incrementally.
+
+Feature flags follow a prefix naming convention that indicate status:
+
+1. `enable-experimental-*`
+2. `enable-v12-*`
```

Comment:
This might be overcomplicating it, but would there ever be a need for an in-between state? Like, the feature is stable and the API will likely not change anymore (but theoretically still could), however it's not committed to the next major version yet?

This could allow for shorter major release cycles as the amount of breaking changes aka v# feature flags is less.

I think my main concern is the psychological aspect of it. In a team with few resources, am I gonna take the *risk* to opt-in to a feature "[...] that the API is likely to change due to [...] instability or incompatibility" if it's not absolutely critical for my product? Then how can we get a decent number of early adopters to test it out as a requirement for promoting it to stable? If there'd be a state in between where we say the API is *un*likely to change and it has been evaluated by a few testers but it's not subject to semver yet, that might lower the barrier.

Just a thought though, I'm torn on this.

Reply:
I've been thinking about this too, am also torn. My main concern is the more "layers" we have, the more likely it is that consumers don't care to understand the naming convention or the slight nuance between them.

> If there'd be a state in between where we say the API is _unlikely_ to change

What if we reframed feature flags to aim for this by default? Flags really shouldn't be highly unstable. The goal is to deliver something we've vetted and are quite confident to fulfill the need or fix an issue. These things are ready for people to explore _today_ with the unlikely possibility of API changes based on user feedback.

We could drop the `experimental` naming to solve part of the psychological aspect too, I think, and just have:

* `enable-*`
* `enable-v#-*`

Reply:
> Like, the feature is stable and the API will likely not change anymore (but theoretically still could), however it's not committed to the next major version yet?

I agree that this should really be our default, and I would say has been for the most part with the flags we already have in place. We've just never outright defined that anywhere. 

I like the idea of dropping the `experimental` part of the flag name, agree that people will be more hesitant to use something with that in the name.

Reply:
I like that plan!